app/claim/route.js

```diff
@@ -1,20 +1,43 @@
 import Ember from 'ember';
 
 export default Ember.Route.extend({
-  unauthenticatedRedirect: 'signup',
+  requireAuthentication: false,
 
   model: function(params){
-    return {
-      verificationCode: params.verification_code,
-      invitationId: params.invitation_id,
-      type: 'invitation'
-    };
+    let invitationId = params.invitation_id;
+    let verificationCode = params.verification_code;
+
+    return this.store.find('invitation', invitationId).then((invitation) => {
+      return { invitation, verificationCode };
+    });
+  },
+
+  setupController(controller, model){
+    let {invitation, verificationCode} = model;
+    controller.set('model', invitation);
+    controller.set('verificationCode', verificationCode);
+  },
+
+  afterModel(model, transition){
+    if (!this.get('session.isAuthenticated')) {
+      this.get('session').attemptedTransition = transition;
+
+      let invitationId = model.invitation.id;
+      let verificationCode = model.verificationCode;
+      this.transitionTo('signup.invitations', invitationId, verificationCode);
```

Comment:
as in make an object with an id and verification code?
i.e.: `transitionTo('signup.invitations', {invitationId: id, verificationCode: code}` ? I have no strong preference either way